### ISSUE TITLE
Bug in reverse AD for matrix determinant operation

### ISSUE THREAD

**Issue by dsyme** [DiffSharp/DiffSharp#42]
When doing 

        let fmsD (x:DM) = x * (log (x / 2.)) |> DM.det

        let xm = DM (Array2D.init 10 10 (fun _ _ -> rnd.NextDouble()))
        computeAdjoints (fmsD (makeReverse 100u xmD))

We get an exception during reverse AD

    Unhandled Exception: System.Exception: Cannot get tangent value of DR.
       at DiffSharp.AD.Float64.D.get_T() in C:\GitHub\dsyme\DiffSharp\src\DiffSharp\AD.Float64.fs:line 111

Basically the use of ``d.T`` (the forward tangent) is failing in this code (note the ``Check this`` comment)

                            | Det_DM(a) -> pushRec ((bx (d.T * d.P * DM.Transpose(DM.Inverse(a))) a) :: t) // Check this

We should not be using the tangent of ``d`` here

**Comment by dsyme** [comment]
I'm closing as this is not relevant in `dev`
